src/ColonesAssetAdapter.ts

```diff
@@ -0,0 +1,145 @@
+
+export type HtlcDetails = OasisHtlc;
+
+export interface OasisClient {
+  getHtlc(id: string): Promise<HtlcDetails>;
+  settleHtlc(id: string, secret: string, settlementJWS: string, authorizationToken?: string): Promise<HtlcDetails>;
+}
+
+export class ColonesAssetAdapter implements AssetAdapter<SwapAsset.CRC> {
+  private cancelCallback: ((reason: Error) => void) | null = null;
+  private stopped = false;
+
+  constructor(public client: OasisClient) { }
+
+  private async findTransaction(
+    id: string,
+    test: (htlc: HtlcDetails) => boolean,
+  ): Promise<HtlcDetails> {
+    const check = async (): Promise<HtlcDetails | null> => {
+      try {
+        const htlc = await this.client.getHtlc(id);
+        if (test(htlc)) return htlc;
+      } catch (error) {
+        console.error(error); // eslint-disable-line no-console
+
+        if ((error as Error).message !== 'HTLC not found') {
+          // TODO: Extract error handling from libswap to application
+          // throw error
+        }
+      }
+      return null;
+    };
+
+    // Check now
+    const htlc = await check();
+    if (htlc) return htlc;
+
+    // Keep checking on an interval
+    return new Promise((resolve, reject) => {
+      const interval = window.setInterval(() => {
+        check().then((htlc) => { // eslint-disable-line @typescript-eslint/no-shadow
+          if (!htlc) return;
+          cleanup();
+          resolve(htlc);
+        });
+      }, 5 * 1000); // Every 5 seconds
+
+      const cleanup = () => {
+        window.clearInterval(interval);
+        this.cancelCallback = null;
+      };
+
+      this.cancelCallback = (reason: Error) => {
+        cleanup();
+        reject(reason);
+      };
+    });
+  }
+
+  public async awaitHtlcFunding(
+    id: string,
+    value: number,
+    data?: string,
+    confirmations?: number,
+    onUpdate?: (htlc: HtlcDetails) => any,
+  ): Promise<HtlcDetails> {
+    return this.findTransaction(
+      id,
+      (htlc) => {
+        // if (htlc.amount + htlc.fee !== value) return false;
+
+        if (htlc.status === HtlcStatus.CLEARED || htlc.status === HtlcStatus.SETTLED) return true;
+
+        if (typeof onUpdate === 'function') onUpdate(htlc);
+
+        return false;
+      },
+    );
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  public async fundHtlc(): Promise<HtlcDetails> {
+    throw new Error('Method "fundHtlc" not available for CRC HTLCs');
+  }
+
+  public async awaitHtlcSettlement(id: string): Promise<OasisHtlc<HtlcStatus.SETTLED>> {
+    return this.findTransaction(
+      id,
+      (htlc) => typeof htlc.preimage.value === 'string',
+    ) as Promise<OasisHtlc<HtlcStatus.SETTLED>>;
+  }
+
+  public async awaitSwapSecret(id: string): Promise<string> {
+    const tx = await this.awaitHtlcSettlement(id);
+    return tx.preimage.value;
+  }
+
+  public async settleHtlc(
+    settlementJWS: string,
+    secret: string,
+    hash: string,
+    authorizationToken?: string,
```

Comment:
The token will have to change to the SMS token, or become the same `tokens?` object as suggested in https://github.com/nimiq/oasis-api-js/pull/1/files#r1625521202.